stix2/datastore/__init__.py

```diff
@@ -542,6 +596,35 @@ def query(self, query=None, _composite_filters=None):
 
         return all_data
 
+    def query_by_type(self, *args, **kwargs):
```

Comment:
It seems like there's a lot of duplication between this function and other functions on the CompositeDataSource. I wonder if there's anything we can do about that. Maybe not in this PR, though.

Reply:
Maybe along the same lines.... How is this any easier than a regular call to `query()`?